python/ray/ml/preprocessors/encoder.py

```diff
@@ -168,7 +168,8 @@ def _transform_pandas(self, df: pd.DataFrame):
         return df
 
     def __repr__(self):
-        return f"<Categorizer columns={self.columns} stats={self.stats_}>"
+        stats = getattr(self, "stats_", None)
```

Comment:
I think we should assign `self.stats_ = None` in the constructor, but I've followed this approach to be consistent with the other `Preprocessor`s in this module.

Side note: PyLint has a warning W0201
> Attribute %r defined outside __init__

Not sure why it's disabled.